pulsar-client/src/main/java/org/apache/pulsar/client/impl/conf/ProducerConfigurationData.java

```diff
@@ -64,6 +64,7 @@
     private int maxPendingMessages = DEFAULT_MAX_PENDING_MESSAGES;
     private int maxPendingMessagesAcrossPartitions = DEFAULT_MAX_PENDING_MESSAGES_ACROSS_PARTITIONS;
     private MessageRoutingMode messageRoutingMode = null;
+    private String messageRouterClassName = null;
```

Comment:
instead having `messageRouterClassName`, we should have factory-class name with param properties so, one can initialize with custom value based on the environemnt.

Reply:
@rdhabalia do you mean something like this?
```
public interface MessageRouterFactory {
    MessageRouter get(Properties properties);
}
```
and while getting message router, we will do something like this:
```
String messageRouterFactoryClassName = conf.getMessageRouterFactoryClassName();
Properties messageRouterFactoryProperties = conf.getMessageRouterFactoryProperties();
clazz = Class.forName(messageRouterFactoryClassName);
MessageRouterFactory messageRouterFactory = (MessageRouterFactory) clazz.getDeclaredConstructor().newInstance();
messageRouter = messageRouterFactory.get(messageRouterFactoryProperties);
```

where `properties` to this factory method will be part of the `ProducerConfigurationData` having name like  `messageRouterFactoryProperties` of type `java.util.Properties` and `messageRouterClassName` should be renamed to `messageRouterFactoryClassName`. This will give more flexibility to the users. If you meant something else, can you please share some example snippet?

Reply:
yes, instead `get()` , it should be `create(..)`